src/main/kotlin/no/skatteetaten/aurora/boober/service/openshift/ResourceMerger.kt

```diff
@@ -48,15 +48,24 @@ private fun updateBuildConfig(mergedResource: JsonNode, existingResource: JsonNo
 private fun updateDeploymentConfig(mergedResource: JsonNode, existingResource: JsonNode) {
     mergedResource.updateField(existingResource, "/spec/triggers/0/imageChangeParams", "lastTriggeredImage")
     val containersField = "/spec/template/spec/containers"
-    val containerCount = (mergedResource.at(containersField) as ArrayNode).size()
 
-    (0 until containerCount).forEach {
-        val containerName = mergedResource.at("$containersField/$it/name").textValue()
+    (mergedResource.at(containersField) as ArrayNode).forEachIndexed { i, containerNode ->
+        val containerName = containerNode.at("/name").textValue()
         val isApplicationContainer = !containerName.endsWith("-sidecar")
 
         // We should allow updates of sidecar container images
         if (isApplicationContainer) {
-            mergedResource.updateField(existingResource, "$containersField/$it", "image")
+            (existingResource.at(containersField) as ArrayNode)
+                .find { it.at("/name").textValue() == containerName }
```

Comment:
Kunne vi tatt dette ut i en funksjon med et beskrivende navn, slik at det er litt klarere hva som skjer og hvorfor

Reply:
Evt extension method på JsonNode for det å finne containerNodeByName også ensureImageIsPreserved